react-validated-input-component/my-app/src/ValidateInput.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+
+export default function ValidateInput() {
+  const [inputValue, setInputValue] = useState('');
+
+  const regex = new RegExp(
+    '(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$'
+  );
+
+  let inputValueLength: number = inputValue.length;
+  let xIHidden: string = 'hidden';
+  let checkHidden: string = 'hidden';
+  let shortPassword: string = 'hidden';
+  let noPassword: string = 'hidden';
+  let regexPass: string = 'hidden';
+
+  if (inputValueLength === 0 || inputValueLength < 8) {
+    xIHidden = '';
+  }
+  if (inputValueLength === 0) {
+    noPassword = '';
+  }
+  if (inputValueLength >= 8 && regex.test(inputValue)) {
+    checkHidden = '';
+  }
+  if (inputValueLength < 8 && inputValueLength > 0) {
+    shortPassword = '';
+  }
+  if (inputValueLength > 8 && !regex.test(inputValue)) {
+    regexPass = '';
+  }
+
+  return (
+    <form
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        padding: '0.5rem',
+        width: '20rem',
+      }}>
+      <label htmlFor="pass">Password</label>
+      <div style={{ display: 'flex', alignItems: 'center' }}>
+        <input
+          id="pass"
+          type="password"
+          value={inputValue}
+          onChange={(e) => setInputValue(e.target.value)}
+        />
+        <i className={`bi bi-x ${xIHidden}`}></i>
+        <i className={`bi bi-check-lg ${checkHidden}`}></i>
+      </div>
+      <p className={`no-pass ${noPassword}`}>A password is required.</p>
+      <p className={`short-pass ${shortPassword}`}>
```

Comment:
There are nice utilities that let you compose class names. You would do this:
`className={classNames('no-pass', noPassword)}`
where the `classNames` function is a utility that concatenates all its truthy arguments.

This would make the above checks even simpler:
```
const regexPass = (inputValueLength > 8 && !regex.test(inputValue)) || 'hidden';
```
This works because if the check is true, `regexPass` gets the value 'hidden', and if it's false, it gets the value `false`, which won't be included by `classNames`.